ui/client/datasets/Annotate.js

```diff
@@ -99,7 +99,7 @@ export default withStyles(({ spacing }) => ({
   };
 
   useEffect(() => {
-    if (!isLoading || (isLoading && !datasetInfo?.id)) {
+    if (!isLoading || (isLoading && !datasetInfo?.id) || isEmpty(annotations?.metadata)) {
```

Comment:
This works in my manual testing, but there are so many edge cases with the annotation page. I'm assuming we're always loading in metadata, but if this isn't the case then we'll need another way of solving this. Without this check the rest of this effect gets run twice, as the annotations object first loads empty, then gets populated. Which means we hit the various endpoints in this useEffect twice (not the end of the world, but better not to).

Reply:
Hm. I'm not sure. As I recall `annotations.metadata` can load empty or populating when getting to this page. And it can be for either:
- Model Output Annotation
- Dataset Register
- Dataset Update

Among others. I'll checkout the branch and try this out so that I can jumpstart my memory and run various tests scenarios.